crates/solver/src/liquidity.rs

```diff
@@ -90,40 +90,57 @@ pub enum Exchange {
 }
 
 /// Used to differentiate between different types of orders that can be sent to solvers.
-/// User orders containing OrderUid are the orders from the orderbook.
+/// User orders (market + limit) containing OrderUid are the orders from the orderbook.
 #[derive(Debug, Clone)]
 #[cfg_attr(test, derive(Derivative))]
 #[cfg_attr(test, derivative(PartialEq))]
-pub enum LimitOrderUid {
-    OrderUid(OrderUid),
+pub enum LimitOrderId {
+    Market(OrderUid),
+    Limit(OrderUid),
+    Liquidity(LiquidityOrderId),
+}
+
+/// Three different types of liquidity orders exist:
+/// 1. Protocol - liquidity orders from the auction model of solvable orders
+/// 2. ZeroEx  - liquidity orders from the zeroex api liquidity collector
+/// 3. Foreign - liquidity orders received as part of the solution from searchers
+///
+/// (1) and (2) are gathered when the auction is cut and they are sent to searchers
+/// (3) are received from searchers as part of the solution.
+#[derive(Debug, Clone)]
+#[cfg_attr(test, derive(Derivative))]
+#[cfg_attr(test, derivative(PartialEq))]
+pub enum LiquidityOrderId {
+    /// TODO: Split into different variants once we have a DTO of order model for `driver` in driver solver colocation
+    /// TODO: The only reason why is together now is because function `normalize_limit_order` can't diferentiate between these two
+    ProtocolOrForeign(OrderUid),
```

Comment:
I would still call this `Protocol` - foreign liquidity orders are still protocol orders, they just don't come from our orderbook.

I think it is fine to punt this until a differentiation is needed (which I can't think of a reason we will need that right now. Maybe there is a solver reward/competition related reason though).

Reply:
I am ok with `Protocol` also since there is a comment explaining. Changed.